internal/codegen/stencil_test.go

```diff
@@ -143,3 +138,63 @@ func TestBadDirReplacement(t *testing.T) {
 	_, err = st.renderDirReplacement("b/c", m, vals)
 	assert.ErrorContains(t, err, "contains path separator in output")
 }
+
+// TestIterations ensures that stencil properly handles iterations based
+// on state, in this case through global variables.
+func TestIterations(t *testing.T) {
+	fs := memfs.New()
+	ctx := context.Background()
+	log := slogext.NewTestLogger(t)
+
+	// create stub manifest
+	f, _ := fs.Create("manifest.yaml")
+	f.Write([]byte("name: testing"))
+	f.Close()
+
+	// create a stub template
+	f, err := fs.Create("templates/test-template.tpl")
```

Comment:
is there a world where the ordering of having this split across 2 files vs 1 is interesting enough to test?

Reply:
I thought about it and no, not really

Reply:
Since files are generated within the same iteration loop it wouldn't change anything